dired-rifle.el

```diff
@@ -94,25 +104,30 @@ With `\\[universal-argument] \\[universal-argument]' the output
 is additionally saved to a buffer named *dired-rifle*.
 
 With a numeric prefix argument ARG, run ARGth rifle rule instead
-of the default one (0th).  The output is discarded."
+of the default one (0th).  The output is discarded.
+
+With 0 as numeric argument, switch between focused file and marked files."
   (interactive "P")
   (let ((inhibit-read-only t))
     (let ((output-buffer (when (equal '(16) arg)
                            "*dired-rifle*"))
-          (path (dired-get-filename)))
-      (let ((program-number (if (consp arg)
-                                (string-to-number
-                                 (replace-regexp-in-string
-                                  "^\\([0-9]+\\).*" "\\1"
-                                  (completing-read "Rifle rule: "
-                                                   (rifle-get-rules path)
-                                                   nil
-                                                   t)))
-                              arg)))
-        (message "Launching rifle...")
-        (rifle-open path
-                    program-number
-                    output-buffer)))))
+          (path (if (equal (equal arg 0) dired-rifle-use-marked-files)
+                    (dired-get-filename)
+                  (dired-get-marked-files))))
+      (dolist (p (flatten-tree path))
```

Comment:
Again, what's the purpose of `flatten-tree` here? Is this to make the output of `(dired-get-filename)` into a one-element list? We can handle it above by doing that explicitly.